modules/GenericDataParser/src/server/generic_data_parser.R

```diff
@@ -706,27 +706,58 @@ validateCalculatedResults <- function(calculatedResults, dryRun, curveNames, tes
         # A project can only be used in the experiment if it's an unrestricted project or belongs to the project which this experiment belongs to
         projectsDT <- data.table(projectDF)
 
-        # Loop through by project code and create a projectAllowedForExperiment column
+        # Add project type if its missing
+        if(!"type" %in% colnames(projectsDT)) {
+          projectsDT[ , type := NA]
+        }
+
+        # Get the project type of the experiment project if required by configurations
+        projectType <- NA
+        if(configList$server.project.type.required) {
+          projectType <- projectsDT[code == projectCode]$type
+          if(is.na(projectType)) {
+            stopUser(paste0("Project '",projectCode,"' does not have a type. Please contact your system administrator."))
+          }
+        }
+
+        # Here we are annotating each of the projects with a boolean as to whether it can be used in the experiment if its used by a lot     
         projectsDT[ , projectAllowedForExperiment := {
-          # Check if the project is unrestricted or if it equals the projectCode of the experiment
-          !isRestricted | code == projectCode
+          # projectType and projectCode are referring to the experiment project
+          # code and isRestricted are referring to the batch project
+          if(is.na(projectType)) {
+            # If the project type is na then this is not a system with project type
+            # so we just check if the project is restricted and if the lot project is the same as the experiment project
+            !isRestricted | code == projectCode
+          } else {
+            # If project type is not na then this is a system with project type
+            switch(projectType,
+                    "RESTRICTED" =  !isRestricted | code == projectCode,
+                    "HYPER_RESTRICTED" = code == projectCode,
+                    "UNRESTRICTED" = TRUE,
+                    "GLOBAL" = ifelse(identical(type, "HYPER_RESTRICTED"), FALSE, TRUE),
+                    stopUser(paste0("Project '",projectCode,"' has an invalid type. Please contact your system administrator."))
+            )
```

Comment:
This is the new logic to compare experiment and batch projects.

Allowed:
  Restricted experiment, unrestricted batch project
  Restricted experiment, matches restricted batch project
  Unrestricted project, not hyper restricted batch project
  Global experiment, not hyper restricted batch project (this is the same as Unrestricted)

Disallowed:
  Restricted experiment, restricted batch project doesn't match
  Hyper restricted experiment, batch project doesn't match regardless of batch restriction
  Global experiment, hyper restricted batch project